minetest/minetest
# Allow non-walkable nodes to destroy falling nodes

This PR allows non-walkable nodes to destroy falling nodes by porting commit [`a512866`](https://gitlab.com/tunnelers-abyss/minetest/-/commit/a51286669cf376a3d9397bc0724a5ed26c41200f) of [Tunneler's Abyss's fork](https://gitlab.com/tunnelers-abyss/minetest/-/commits/tunnelers-5.9dev). Nodes with group `destroy_falling_node = 1` will not be displaced by falling nodes; instead, if falling nodes attempt to create a node here, they will be destroyed and dropped as items. If you play a certain Mojang game, this copies the behavior of torches destroying falling sands and gravel.

Starting from cherry-picking Tunneler's commit, I changed the group name from a weird `nostomp` to a more informative one, `destroy_falling_node`. Documentation and testing nodes are also added. _Note to Tunneler's devs:_ The group name has to be changed after merging this (likely in 5.10).

Technically all nodes can destroy falling nodes, but this flag is only practically useful on non-walkable nodes, hence the name.

## To do

This PR is Ready for Review. This code has already proven to work and is stable on Tunneler's Abyss.

## How to test

Try to fall nodes onto the following nodes:
1. `testnodes:destroy_falling_node_0`: Should be displaced
2.  `testnodes:destroy_falling_node_1`: Should destroy the falling node
3. Any existing non-walking node (I use a light source): Should be displaced

This is my test footage (the signs are de facto game-independent so no worries): 

https://github.com/minetest/minetest/assets/55009343/b8ed7b95-72fb-4c69-afd2-a5897c7bb2c7
